src/Entities/Boat.cpp

```diff
@@ -142,11 +142,14 @@ void cBoat::Tick(std::chrono::milliseconds a_Dt, cChunk & a_Chunk)
 		return;
 	}
 
-	if (IsBlockWater(m_World->GetBlock(POSX_TOINT, POSY_TOINT, POSZ_TOINT)))
+	// A real boat floats.
+	// Propel to top water block and sit slightly beneath the waterline:
+	if (IsBlockWater(m_World->GetBlock(POSX_TOINT, POSY_TOINT, POSZ_TOINT)) && ((GetPosY() - POSY_TOINT) < 0.6))
 	{
 		if (GetSpeedY() < 2)
 		{
-			AddSpeedY(0.2);
+			const auto DtSec = std::chrono::duration_cast<std::chrono::duration<double>>(a_Dt).count();
+			SetSpeedY((-m_Gravity + 1) * DtSec);
```

Comment:
Can you add a comment explaining this magic constant - it's not intuitive that at boat should float upwards at the magnitude of "gravity plus a bit".